### PR TITLE
*: fix prewrite secondary rows missing data bug

### PR DESCRIPTION
**Pull request by c4pt0r**
and add some hooks for testing.

### CODE REVIEW

**Comment by c4pt0r** [comment]
@shenli @siddontang  PTAL

File: themis_txn.go [hunk 1]
```diff
@@ -185,17 +191,22 @@ func (txn *themisTxn) Commit() error {
...
 		return kv.ErrRetryable
 	}
-
 	txn.commitSecondary()
+	log.Debug("themis txn commit successfully", txn.startTs, txn.commitTs)
 	return nil
 }
 
 func (txn *themisTxn) commitSecondary() {
+	if txn.hooks.beforeCommitSecondary != nil {
```
Comment (siddontang) [inline themis_txn.go]: you can use a default hook doing nothing and customize special test hook.

File: txn_test.go [hunk 2]
```diff
@@ -378,3 +367,127 @@ func (s *TransactionTestSuit) TestExceedMaxRows(c *C) {
...
+
+		return true, nil, nil
+	}
+
+	tx1 := newTxn(s.cli, conf)
+	tx1.(*themisTxn).setHook(hook)
+	tx1.Put(themisTestTableName, hbase.NewPut([]byte("A")).AddValue(cf, q, []byte("A")))
+	tx1.Put(themisTestTableName, hbase.NewPut([]byte("B")).AddValue(cf, q, []byte("B")))
+	tx1.Put(themisTestTableName, hbase.NewPut([]byte("C")).AddValue(cf, q, []byte("C")))
+	tx1.Commit()
```
Comment (siddontang) [inline txn_test.go]: no need to check err for commit?

File: themis_lock_manager.go [hunk 3]
```diff
@@ -93,7 +93,8 @@ func (m *themisLockManager) GetCommitTimestamp(cc *hbase.ColumnCoordinate, prewr
 	// add del write column
 	g.AddStringColumn("#d", qual)
 	// time range => [ours startTs, +Inf)
-	g.AddTimeRange(prewriteTs, math.MaxInt64)
+	g.AddTimeRange(prewriteTs, math.MaxInt32)
```
Comment (shenli) [inline themis_lock_manager.go]: Java Hbase client use long as timerange parameter:
https://hbase.apache.org/apidocs/org/apache/hadoop/hbase/client/Get.html#setTimeRange(long,%20long)

long is a 64bits integer: https://docs.oracle.com/javase/tutorial/java/nutsandbolts/datatypes.html

Why change this ftom math.MaxInt64 to MaxInt32?

File: txn_test.go [hunk 4]
```diff
@@ -378,3 +367,128 @@ func (s *TransactionTestSuit) TestExceedMaxRows(c *C) {
...
+					txn.secondaryRows = append(txn.secondaryRows, rowMutation)
+				}
+			}
+		}
+		return true, nil, nil
+	}
+	return hook
+}
+
+func (s *TransactionTestSuit) TestPrewriteSecondaryMissingRows(c *C) {
```
Comment (shenli) [inline txn_test.go]: Please add a few description about the bug.
It is not easy for others to figure out the point.

**Comment by ngaut** [comment]
LGTM

File: txn_test.go (hunk 4 above)
Comment (c4pt0r) [inline txn_test.go]: https://github.com/pingcap/go-themis/issues/19

**Pull request merged**